File: tunnel-client/src/main/java/com/tuuzed/tunnel/client/internal/AttributeKeys.java
package com.tuuzed.tunnel.client.internal;

import com.tuuzed.tunnel.client.TunnelClientDescriptor;
import com.tuuzed.tunnel.common.proto.ProtoRequest;
import io.netty.channel.Channel;
import io.netty.util.AttributeKey;
import org.jetbrains.annotations.NotNull;

public final class AttributeKeys {

    @NotNull
    public static final AttributeKey<Long> SESSION_TOKEN = AttributeKey.newInstance("$SESSION_TOKEN");

    @NotNull
    public static final AttributeKey<Channel> NEXT_CHANNEL = AttributeKey.newInstance("$NEXT_CHANNEL");

    @NotNull
    public static final AttributeKey<Long> TUNNEL_TOKEN = AttributeKey.newInstance("$TUNNEL_TOKEN");

    @NotNull
    public static final AttributeKey<ProtoRequest> PROTO_REQUEST = AttributeKey.newInstance("$PROTO_REQUEST");

    @NotNull
    public static final AttributeKey<Boolean> FATAL_FLAG = AttributeKey.newInstance("$FATAL_FLAG");

    @NotNull
    public static final AttributeKey<Throwable> FATAL_CAUSE = AttributeKey.newInstance("$FATAL_CAUSE");


    public static final AttributeKey<TunnelClientDescriptor> TUNNEL_CLIENT_DESCRIPTOR = AttributeKey.newInstance("$TUNNEL_CLIENT_DESCRIPTOR");


}

File: tunnel2-client/src/main/kotlin/tunnel2/client/internal/_AttributeKeys.kt
@file:JvmName("_AttributeKeys")
@file:Suppress("HasPlatformType")

package tunnel2.client.internal

import io.netty.channel.Channel
import io.netty.util.AttributeKey
import tunnel2.client.TunnelClientDescriptor
import tunnel2.common.TunnelRequest

val AK_TUNNEL_ID = AttributeKey.newInstance<Long>("AK_TUNNEL_TOKEN")

val AK_SESSION_ID = AttributeKey.newInstance<Long>("AK_SESSION_ID")

val AK_NEXT_CHANNEL = AttributeKey.newInstance<Channel>("NEXT_CHANNEL")

val AK_TUNNEL_REQUEST = AttributeKey.newInstance<TunnelRequest>("AK_TUNNEL_REQUEST")

val AK_ERR_FLAG = AttributeKey.newInstance<Boolean>("AK_ERR_FLAG")

val AK_ERR_CAUSE = AttributeKey.newInstance<Throwable>("AK_ERR_CAUSE")

val AK_TUNNEL_CLIENT_DESCRIPTOR = AttributeKey.newInstance<TunnelClientDescriptor>("AK_TUNNEL_CLIENT_DESCRIPTOR")
